tests/test_kubernetes.py

```diff
@@ -114,6 +114,10 @@ def test_get_job(
     _check_resource_requirements(
         container, memory_args.memory_gb, memory_args.memory_gb_limit, cpu_count
     )
+    toleration = pod_spec.tolerations[0]
+    assert toleration.effect == "NoSchedule"
+    assert toleration.key == "dedicated"
+    assert toleration.value == "climate-sim-pool"
```

Comment:
This would be suitable to put in a `_check_tolerations` helper function, and should additionally test that `len(pod_spec.tolerations) == 1`. After that, looks ready to merge.

Reply:
Thanks! Changes have been made to address these two suggestions.